src/validation/generator.jl

```diff
@@ -0,0 +1,25 @@
+
+"""Validates the contents of generators."""
+function validate(gen::GenClasses)::Bool
+    is_valid = true
+
+    for field in fieldnames(typeof(gen))
+        generators = getfield(gen, field)
+        if !isnothing(generators)
+            if !validate_devices(generators)
+                is_valid = false
+            end
+        end
+    end
+
+    @debug "GenClasses validation" is_valid
+    return is_valid
+end
+
+"""Validates the contents of a Generator."""
+function validate(generator::Generator)::Bool
+    is_valid = true
+
+    @debug "Generator validation" generator.name is_valid
+    return is_valid
+end
```

Comment:
Is this incomplete? This function just returns `true`.

Reply:
Intentionally incomplete.  I wanted to get agreement on the design before including details.  This is roughly how it will look.  Another missing item is a config file containing default values, mins, and maxes.

Reply:
Add a comment that says this will be implemented later.

Reply:
There is a comment in the System constructor.  Does that work for you?